parlai/tasks/glue/agents.py

```diff
@@ -15,5 +30,116 @@ class ColaTeacher(AbstractHuggingFaceTeacher):
     hf_splits_mapping = {'train': 'train', 'valid': 'validation', 'test': 'test'}
 
 
-class DefaultTeacher(ColaTeacher):
+class MnliTeacher(AbstractHuggingFaceTeacher):
+    hf_path = 'glue'
+    hf_name = 'mnli'
+    hf_text_fields = ['premise', 'hypothesis']
+    hf_label_field = 'label'
+    hf_splits_mapping = {
+        'train': 'train',
+        'valid': 'validation_matched',
+        'test': 'test_matched',
+    }
+
+
+class MnliMatchedTeacher(AbstractHuggingFaceTeacher):
+    """
+    Note: this is an evaluation dataset so it only has valid and test splits
+    """
+
+    hf_path = 'glue'
+    hf_name = 'mnli_matched'
+    hf_text_fields = ['premise', 'hypothesis']
+    hf_label_field = 'label'
+    hf_splits_mapping = {'train': None, 'valid': 'validation', 'test': 'test'}
+
+
+class MnliMismatchedTeacher(AbstractHuggingFaceTeacher):
+    """
+    Note: this is an evaluation dataset so it only has valid and test splits
+    """
+
+    hf_path = 'glue'
+    hf_name = 'mnli_mismatched'
+    hf_text_fields = ['premise', 'hypothesis']
+    hf_label_field = 'label'
+    hf_splits_mapping = {'train': None, 'valid': 'validation', 'test': 'test'}
```

Comment:
added these 2 as consistency since on hugging face they also have them as separate datasets but technically you can access the data from the mnli dataset splits ('validation_mismatched', 'validation_matched', and the same for test).

Reply:
Great, just checked the leaderboard and those are separate numbers, this seems good